postfix/map.jinja

```diff
@@ -1,36 +1,81 @@
-{% set postfix = salt['grains.filter_by']({
+# -*- coding: utf-8 -*-
+# vim: ft=jinja
+
+{%- macro deep_merge(a, b) %}
```

Comment:
Not that I disagree but why not use the defaults, oscodenamemap, osmap, etc, stacking like appears to become common in other formulas such as https://github.com/saltstack-formulas/squid, https://github.com/saltstack-formulas/openvpn and https://github.com/saltstack-formulas/postgres-formula ?

Reply:
I've just didn't look any further after finding this macro in the salt-formula.
The stacking you mention *is* nicer than this macro...

Reply:
I'll wait for @aboe76 or someone else feedback but imho we should avoid such complex macro if not strictly needed.

Reply:
This jinja2 is an workaround and shouldn't be used if it isn't necessary...or avoidable...
I would rather see defaults.yaml, oscodenamemap, and osmap, where possible because those constructions are extendable...without too much knowledge about jinja.

Reply:
Should be easy to swap this part out or add it on master and cherrypick my other changes. We'll see if I can get to this tomorrow.